grid-cli/internal/cmd/get.go

```diff
@@ -24,17 +24,16 @@ func GetVM(ctx context.Context, t deployer.TFPluginClient, name string) (workloa
 	// try to get contracts with the new project name format "vm/<name>"
 	projectName := fmt.Sprintf("vm/%s", name)
 
-	nodeContractIDs, err := t.ContractsGetter.GetNodeContractsByTypeAndName(projectName, workloads.VMType, name)
-	if err != nil {
-		return workloads.Deployment{}, err
-	}
+	var networkContractIDs map[uint32]uint64
 
-	networkContractIDs, err := t.ContractsGetter.GetNodeContractsByTypeAndName(projectName, workloads.NetworkType, fmt.Sprintf("%snetwork", name))
-	if err != nil {
-		return workloads.Deployment{}, err
-	}
+	nodeContractIDs, err := t.ContractsGetter.GetNodeContractsByTypeAndName(projectName, workloads.VMType, name)
+	if err == nil {
+		networkContractIDs, err = t.ContractsGetter.GetNodeContractsByTypeAndName(projectName, workloads.NetworkType, fmt.Sprintf("%snetwork", name))
+		if err != nil {
+			return workloads.Deployment{}, err
+		}
 
-	if len(nodeContractIDs) == 0 && len(networkContractIDs) == 0 {
+	} else {
```

Comment:
this doesn't tell the difference between an error with fetching/parsing or just no contracts found, I think it shouldn't be handled that way as we only should continue if there are no errors but we didn't find a contract. We can do that with either defining an error in the function we call that we can check on it here or stop it from returning errors on 0 contracts. I prefer defining an error as to not change it's functionality.

Reply:
I've added a new error type ` ErrorContractNotFound` in case of empty list of contracts and used it instead of just checking if error is not nil